sklearn/discriminant_analysis.py

```diff
@@ -494,6 +494,7 @@ def _solve_svd(self, X, y):
         rank = np.sum(S > self.tol)
         # Scaling of within covariance is: V' 1/S
         scalings = (Vt[:rank] / std).T / S[:rank]
+        fac = 1.0 / (n_classes - 1)
```

Comment:
There isnt' really a "between class covariance" when there is only one class. If we must maintain backward compatibility, we can set `fac=1` here, which results in `X` zeros because `self.means_ - self.xbar_` are all zeros. (With one class `self.priors_` is all ones)

Reply:
Implemented this solution, thanks!